.github/workflows/codeql.yaml

```diff
@@ -0,0 +1,42 @@
+name: "CodeQL"
+
+on:
+  schedule:
+    - cron: '0 13 * * 1' # At 1:00 PM UTC every Monday
+  pull_request:
+    paths:
+      - '.github/workflows/codeql.yaml'
```

Comment:
Do we want this to run on every PR?

Reply:
In the future that is the goal.  However right now I don't want people to burden themselves with these alerts until I am able to triage and make sure that we don't have a lot of false positive noise first.

But if we decide CodeQL is a good option going forward, I would like to have the feedback provided early, at the PR.